File: test/fixtures/php_project/src/Storage.php
<?php

declare(strict_types=1);

namespace LspcmdFixture;

/**
 * Interface for user storage backends.
 */
interface Storage
{
    /**
     * Saves a user to the storage.
     *
     * @param User $user The user to save
     */
    public function save(User $user): void;

    /**
     * Loads a user by email.
     *
     * @param string $email The email to search for
     * @return User|null The user if found, null otherwise
     */
    public function load(string $email): ?User;

    /**
     * Deletes a user by email.
     *
     * @param string $email The email of the user to delete
     * @return bool True if the user was deleted
     */
    public function delete(string $email): bool;

    /**
     * Lists all users.
     *
     * @return User[] All users in the storage
     */
    public function list(): array;
}

File: test/unit/fixtures/java_project/src/main/java/com/example/Storage.java
package com.example;

import java.util.List;

/**
 * Interface defining the storage operations for users.
 */
public interface Storage {
    /**
     * Saves a user to storage.
     *
     * @param user The user to save
     */
    void save(User user);

    /**
     * Loads a user by email address.
     *
     * @param email The email to look up
     * @return The user if found, null otherwise
     */
    User load(String email);

    /**
     * Deletes a user by email address.
     *
     * @param email The email of the user to delete
     * @return true if the user was deleted
     */
    boolean delete(String email);

    /**
     * Lists all stored users.
     *
     * @return A list of all users
     */
    List<User> list();
}
